pkg/http/server_test.go

```diff
@@ -0,0 +1,108 @@
+package http_test
```

Comment:
I think we don't need a separate package for test cases, right?

Reply:
I usually do that to make sure to only test the public surface of a package.